static/js/services/search.js

```diff
@@ -1,125 +1,70 @@
-      // responses = [searchResults1, searchResult2, ...]
-      var getIntersection = function(responses) {
-        var intersection = responses.pop();
-        intersection = _.uniq(intersection, 'id');
-        _.each(responses, function(response) {
-          intersection = _.reject(intersection, function(row) {
-            return !_.findWhere(response, { id: row.id });
-          });
-        });
-        return intersection;
-      };
-
-      // Queries view as specified by request object coming from GenerateSearchQueries.
-      // request = {view, union: true, paramSets: [params1, ...] }
-      // or
-      // request = {view, params: {...} }
-      var queryView = function(request) {
-        var paramSets = request.union ? request.paramSets : [ request.params ];
-        return $q.all(paramSets.map(function(params) {
-          return DB().query(request.view, params);
-        }))
-          .then(function(data) {
-            return _.flatten(data.map(function(datum) {
-              return datum.rows;
-            }), true);
-          });
-      };
-
-      var queryViewPaginated = function(request, options) {
-        request.params = request.params || {};
-        request.params.limit = options.limit;
-        request.params.skip = options.skip;
-        return queryView(request);
-      };
-
-      var getRows = function(type, requests, options) {
-        if (requests.length === 1 && requests[0].ordered) {
-          // 1 ordered view - let the db do the pagination for us
-          return queryViewPaginated(requests[0], options);
-        }
-        // multiple requests - have to manually paginate
-        return $q.all(requests.map(queryView))
-          .then(getIntersection)
-          .then(_.partial(getPageRows, type, _, options));
-      };
-
-      var generateRequests = function(type, filters, options) {
-        var requests = GenerateSearchRequests(type, filters);
-        if (!options.force && debounce(type, filters, requests)) {
-          return [];
-        }
-        return requests;
-      };
+      var _search = SearchFactory();
 
       return function(type, filters, options) {
         options = options || {};
         _.defaults(options, {
           limit: 50,
           skip: 0
         });
-        return $q.resolve(generateRequests(type, filters, options))
-          .then(function(requests) {
-            return getRows(type, requests, options);
-          })
+
+        // Gareth: I have maintained the original debouncing code, but I'm
+        // not sure it actually does anything useful? Shouldn't it attach to an
+        // ongoing result (promise) and not just return []?
+        if (!options.force && debounce(type, filters, options)) {
+          return $q.resolve([]);
+        }
+
+        return _search(type, filters, options)
           .then(function(results) {
-            return GetDataRecords(_.pluck(results, 'id'), options);
+            return GetDataRecords(results, options);
```

Comment:
What happened to the plucking?

Reply:
The extract search service (which in webapp and this PR is only used here) has a slightly different api: it now returns an array of ids, so no plucking necessary!